src/Scheduler.scala

```diff
@@ -0,0 +1,32 @@
+import java.util.concurrent.{Executors, PriorityBlockingQueue}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait Scheduler {
+  def schedule[T](task: () => T, timestamp: Long): Future[T]
+  def start()(implicit ec: ExecutionContext): Unit
+}
+
+
+class SchedulerImp(val minHeap: PriorityBlockingQueue[ScheduledTask[_]]) extends Scheduler {
+
+  // ExecutionContext for Future
+  implicit val ec: ExecutionContext = ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(5))
```

Comment:
We should use the same ExecutionContext in the Main class. For maintainability, it's always better to have only one `Thread Pool`.